core/src/main/java/org/opensearch/sql/expression/function/BuiltinFunctionRepository.java

```diff
@@ -182,6 +182,9 @@ private FunctionBuilder getFunctionBuilder(
     if (isCastFunction(functionName) || sourceTypes.equals(targetTypes)) {
       return funcBuilder;
     }
+    if (functionResolverMap.get(functionName) instanceof VarargsFunctionResolver) {
+      return castArguments(sourceTypes, sourceTypes, funcBuilder);
```

Comment:
sourceTypes twice?

Reply:
@acarbonetto,
see https://github.com/Bit-Quill/opensearch-project-sql/pull/200#discussion_r1062986070

Reply:
I think a code comment is required to make things clear